persistence/src/test/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJobIT.java

```diff
@@ -0,0 +1,268 @@
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+import edu.unc.lib.dl.acl.util.AccessGroupSet;
+import edu.unc.lib.dl.acl.util.GroupsThreadStore;
+import edu.unc.lib.dl.fcrepo4.CollectionObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.PremisEventObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.TransactionManager;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.search.solr.model.ObjectPath;
+import edu.unc.lib.dl.search.solr.service.ObjectPathFactory;
+import edu.unc.lib.dl.sparql.SparqlUpdateService;
+import edu.unc.lib.dl.test.RepositoryObjectTreeIndexer;
+import edu.unc.lib.dl.test.TestHelper;
+/**
+ *
+ * @author harring
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextHierarchy({
+    @ContextConfiguration("/spring-test/test-fedora-container.xml"),
+    @ContextConfiguration("/spring-test/cdr-client-container.xml"),
+})
+public class DestroyObjectsJobIT {
+
+    @Autowired
+    private RepositoryObjectFactory repoObjFactory;
+    @Autowired
+    private RepositoryObjectLoader repoObjLoader;
+    @Autowired
+    private TransactionManager txManager;
+    @Autowired
+    @Spy
+    private DestroyProxyService spyProxyService;
+    @Autowired
+    private ObjectPathFactory pathFactory;
+    @Mock
+    private ObjectPath path;
+    @Autowired
+    private SparqlUpdateService sparqlUpdateService;
+    @Autowired
+    private Model queryModel;
+    @Autowired
+    private FcrepoClient fcrepoClient;
+
+    private RepositoryObjectTreeIndexer treeIndexer;
+
+    private List<PID> objsToDestroy = new ArrayList<>();
+
+    private DestroyObjectsJob job;
+
+    @Before
+    public void init() throws Exception {
+        initMocks(this);
+        TestHelper.setContentBase("http://localhost:48085/rest");
+        GroupsThreadStore.storeUsername("test_user");
+        GroupsThreadStore.storeGroups(new AccessGroupSet("adminGroup"));
+
+        treeIndexer = new RepositoryObjectTreeIndexer(queryModel, fcrepoClient);
+
+        objsToDestroy = createContentTree();
+
+        when(pathFactory.getPath(any(PID.class))).thenReturn(path);
+        when(path.toNamePath()).thenReturn("path/to/object");
+    }
+
+    @Test
+    public void destroySingleObjectTest() {
+        PID fileObjPid = objsToDestroy.get(2);
+        initializeJob(Arrays.asList(fileObjPid));
+
+        job.run();
+
+        verify(spyProxyService).destroyProxy(fileObjPid);
+        FileObject fileObj = repoObjLoader.getFileObject(fileObjPid);
+        assertTrue(fileObj.getModel().contains(fileObj.getResource(), RDF.type, Cdr.Tombstone));
```

Comment:
It would be good to verify that the tombstone retains the original rdf type as well in at least one of these tests.

Reply:
good idea. done!